### PR TITLE
enable simulation clock for timer canceling test.

### PR DESCRIPTION
**Pull request by fujitatomoya**
replacement for https://github.com/ros2/rclcpp/pull/2453

this depends on https://github.com/ros2/rclcpp/pull/2456

### CODE REVIEW

**Comment by fujitatomoya** [comment]
CC: @alsora @Crola1702

File: rclcpp/test/rclcpp/executors/executor_types.hpp [hunk 1]
```diff
@@ -32,6 +32,12 @@ using ExecutorTypes =
   rclcpp::executors::StaticSingleThreadedExecutor,
   rclcpp::experimental::executors::EventsExecutor>;
 
+using MainExecutorTypes =
+  ::testing::Types<
+  rclcpp::executors::SingleThreadedExecutor,
+  rclcpp::executors::MultiThreadedExecutor,
+  rclcpp::executors::StaticSingleThreadedExecutor>;
```
Comment (clalancette) [inline rclcpp/test/rclcpp/executors/executor_types.hpp:39]: Given that this is only used in `test_executors_timer_cancel_behavior.cpp` for now, I think we can just move it there.  If it turns out we need it in other tests, we can always move it later.

**Review by clalancette — CHANGES_REQUESTED** [review]
One thing I think we should change, this otherwise looks good to me.

**Comment by fujitatomoya** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20516/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14916/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21263/)

**Review by clalancette — APPROVED** [review]

**Comment by fujitatomoya** [comment]
yaml_cpp_vendor is failing during clone, i am not sure what the cause this is. retry CI.

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/20520/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/21270/)

**Comment by fujitatomoya** [comment]
https://ci.ros2.org/job/ci_windows/21270/testReport/ is unrelated.

**Pull request merged**